src/Backend/Modules/Pages/Domain/ModuleExtra/ModuleExtraRepository.php

```diff
@@ -97,4 +97,30 @@ public function getWidgetDataByModuleAndActionAndItemId(string $module, string $
 
         return $results[0]['data'];
     }
+
+    public function getWidgetId(string $module, string $action): ?int
+    {
+        /** @var ModuleExtra|null $moduleExtra */
+        $moduleExtra = $this
+            ->createQueryBuilder('em')
+            ->where('em.module = :module')
+            ->andWhere('em.type = :type')
+            ->andWhere('em.action = :action')
+            ->setParameters(
+                [
+                    'module' => $module,
+                    'type' => (string) ModuleExtraType::widget(),
+                    'action' => $action,
+                ]
+            )
+            ->setMaxResults(1)
+            ->getQuery()
+            ->getOneOrNullResult();
```

Comment:
add `->select('em.id')` and use `getSingleScalarResult` instead of `getOneOrNullResult`